docs/querying/filters.md

```diff
@@ -37,261 +37,219 @@ Apache Druid supports the following types of filters.
+| `value` | String value to match. | No, if not specified the filter will match NULL values. |
+| `extractionFn` | [Extraction function](./dimensionspecs.md#extraction-functions) to apply to `dimension` prior to value matching. See [Filtering with Extraction Functions](#filtering-with-extraction-functions) for details. | No |
 
-``` json
-"filter": { "type": "selector", "dimension": <dimension_string>, "value": <dimension_value_string> }
-```
-
-This is the equivalent of `WHERE <dimension_string> = '<dimension_value_string>'` or `WHERE <dimension_string> IS NULL`
-(if the `value` is `null`).
+The selector filter is limited to only being able to match against `STRING` (single and multi-valued), `LONG`, `FLOAT`,
+`DOUBLE` types. Use the newer null and equality filters to match against `ARRAY` or `COMPLEX` types.
 
-The selector filter supports the use of extraction functions, see [Filtering with Extraction Functions](#filtering-with-extraction-functions) for details.
+When the selector filter matches against numeric inputs, the string `value` will be best effort coerced into a numeric value.
 
-## Column comparison filter
-
-The column comparison filter is similar to the selector filter, but instead compares dimensions to each other. For example:
+### Example: equivalent of `WHERE someColumn = 'hello'`
 
 ``` json
-"filter": { "type": "columnComparison", "dimensions": [<dimension_a>, <dimension_b>] }
+{ "type": "selector", "dimension": "someColumn", "value": "hello" }
 ```
 
-This is the equivalent of `WHERE <dimension_a> = <dimension_b>`.
 
-`dimensions` is list of [DimensionSpecs](./dimensionspecs.md), making it possible to apply an extraction function if needed.
-
-## Regular expression filter
-
-The regular expression filter is similar to the selector filter, but using regular expressions. It matches the specified dimension with the given pattern. The pattern can be any standard [Java regular expression](http://docs.oracle.com/javase/6/docs/api/java/util/regex/Pattern.html).
+### Example: equivalent of `WHERE someColumn IS NULL`.
 
 ``` json
-"filter": { "type": "regex", "dimension": <dimension_string>, "pattern": <pattern_string> }
+{ "type": "selector", "dimension": "someColumn", "value": null }
 ```
 
-The regex filter supports the use of extraction functions, see [Filtering with Extraction Functions](#filtering-with-extraction-functions) for details.
 
+## Equality Filter
 
-## Logical expression filters
+The equality filter is a replacement for the selector filter with the ability to match against any type of column. The equality filter intends to have more SQL compatible behavior than the selector filter and so cannot match NULL values, use the null filter instead.
 
-### AND
+Druid's SQL planner will by default use the equality filter instead of selector filter whenever `druid.generic.useDefaultValueForNull=false`, or if `sqlUseBoundAndSelectors` is set to false on the [SQL query context](./sql-query-context.md).
```

Comment:
```suggestion
Druid's SQL planner uses the equality filter by default instead of selector filter whenever `druid.generic.useDefaultValueForNull=false`, or if `sqlUseBoundAndSelectors` is set to false on the [SQL query context](./sql-query-context.md).
```